code/modules/admin/buildmode.dm

```diff
@@ -234,6 +256,33 @@
 			generator_path = type
 			cornerA = null
 			cornerB = null
+		if(AREAEDIT_BUILDMODE)
+			var/target_path = input(user,"Enter typepath:", "Typepath", "/area")
+			var/areatype = text2path(target_path)
+			if(ispath(areatype,/area))
+				var/areaname = input(user,"Enter area name:", "Area name", "Area")
+				if(!areaname || !length(areaname))
+					return
+				storedarea = new areatype
+				storedarea.power_equip = 0
+				storedarea.power_light = 0
+				storedarea.power_environ = 0
+				storedarea.always_unpowered = 0
+				storedarea.name = areaname
+				areaimage.loc = storedarea
+		if(FILL_BUILDMODE)
+			var/target_path = input(user,"Enter typepath:" ,"Typepath","/obj/structure/closet")
+			objholder = text2path(target_path)
+			if(!ispath(objholder))
+				objholder = pick_closest_path(target_path)
+				if(!objholder || ispath(objholder, /area))
+					objholder = /obj/structure/closet
+					return
+			else
+				if(ispath(objholder,/mob) && !check_rights(R_DEBUG,0))
+					objholder = /obj/structure/closet
+			cornerA = null
+			cornerB = null
 
```

Comment:
You may want to use `pick_closest_path` so they don't need to give the exact path to make something.